cpp/dolfinx/fem/assemble_vector_impl.h

```diff
@@ -945,18 +1029,23 @@ void assemble_vector(
     std::span<const std::int32_t> cells = L.domain(IntegralType::cell, i);
     if (bs == 1)
     {
-      impl::assemble_cells<T, 1>(dof_transform, b, x_dofmap, x, cells, dofs, bs,
-                                 fn, constants, coeffs, cstride, cell_info);
+      impl::assemble_cells<T, 1>(
+          dof_transform, b, x_dofmap, x, cells,
+          {dofs, bs, L.domain(IntegralType::cell, i, *mesh0)}, fn, constants,
+          coeffs, cstride, cell_info0);
```

Comment:
Ok, so I see that you are sending in cell_info for the submesh here. Maybe we should change the name of the cell_info inside each of these functions to ensure that we understand what they are associated with? 

I think we need proper docstrings of all the assemblies (`assemble_cells`, `assemble_exterior_facets`, ..., `lift_bc_cells` etc) as they are getting more complicated with this kind of additional arguments and association with different meshes.